File: content/realm/examples/java/sync/app/src/main/java/com/mongodb/realm/examples/model/java/Project.java
package com.mongodb.realm.examples.model.java;
// :snippet-start: project

import org.bson.types.ObjectId;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmClass;
import io.realm.annotations.Required;

// :remove-start:
@RealmClass(name = "DefinitelyNotProject")
// :remove-end:
public class Project extends RealmObject {
    @PrimaryKey
    public ObjectId _id;
    @Required
    public String name;
    public RealmList<ProjectTask> tasks = new RealmList<>();
}
// :snippet-end:

File: content/realm/examples/java/sync/app/src/main/java/com/mongodb/realm/examples/model/Project.kt
package com.mongodb.realm.examples.model
// :snippet-start: project
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required
import org.bson.types.ObjectId

open class Project(
    @PrimaryKey
    var _id: ObjectId = ObjectId(),
    @Required
    var name: String = "",
    var tasks: RealmList<ProjectTask> = RealmList(),
): RealmObject()
// :snippet-end: